### DISCUSSION TITLE
Need urgent help regarding nginx(npm)  doh

### DISCUSSION THREAD

**Discussion by Berrus** [AdguardTeam/AdGuardHome#4483] [Q&A]
I'm running adguard home on docker along with nginx (npm) (also on docker) hosted on vps(linux-Ubuntu 20.xx).

I would like deploy DOH through nginx(npm) but I cannot find a clear solution (noob solution) 

-have enabled untrusted doh from adguard config file
-port 443 is taken by nginx, thus doh is hosted on port 444
-also added custom domain(certificate) on both nginx and adguard to enable https

**Comment by ainar-g** [comment]
We have an [FAQ entry][1] regarding reverse proxies on our Wiki.  Does that work for you?  If no, what is your Nginx configuration and how do you check the connectivity?

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#reverseproxy

**Reply by Berrus** [reply]
So i tried but didn't work,

On nginx I can access adguard dashboard fine port(3000) ![image](https://user-images.githubusercontent.com/50305674/162979417-eb367123-e58b-4887-b67b-b6323d6d41ec.jpeg) 

Above image is the config of my nginx on for doh server that is running on adguard. I tried putting thing as mention FAQ in advanced tab of this but could not get working.

When I tried entering the server on AdGuard on ios, it keep giving error invalid upstream.

**Reply by ainar-g** [reply]
@Berrus, unfortunately I'm not familiar with any GUI utilities for Nginx configuration.  Is the Nginx server on the same machine as AdGuard Home?  If so, perhaps try using `127.0.0.1` as the IP address?
